_drafts/gdbundle.md

```diff
@@ -0,0 +1,549 @@
+
+#### Modification to `.gdbinit`
+
+Just like Vim plugin managers require the user to edit their `~/.vimrc`,
+gdbundle requires a snippet of code to be added to a loaded `.gdbinit` file.
+
+```
+# -- GDBUNDLE_EDITS_BEGIN
+python
+
+import os,subprocess,sys
+# Execute a Python using the user's shell and pull out the sys.path (for site-packages)
+paths = subprocess.check_output('python -c "import os,sys;print(os.linesep.join(sys.path).strip())"',shell=True).decode("utf-8").split()
+# Extend GDB's Python's search path
+sys.path.extend(paths)
+
+import gdbundle
+gdbundle.init()
+
+end
+# -- GDBUNDLE_EDITS_END
+```
+
+For more information about exactly what is happening above, feel free to read
+the short section in
+[Using PyPi Packages with GDB](https://interrupt.memfault.com/blog/using-pypi-packages-with-GDB#setting-syspath-within-gdbinit).
+
+The first block **is required** to be able to use Python packages installed in
+the activated virtual environment, Conda environment, or non-default Python
+installation.
+
+The second part, `gdbundle.init()`, is what is new and will load installed
+gdbundle plugins.
+
+> NOTE: The major (and ideally minor) version of Python linked with GDB should
+> match the version that is activated at the time of launch, whether that is a
+> virtual environment, Conda environment, `brew` Python installation, etc.
+> Mismatched major versions will cause packages and plugins to either break in
+> mysterious ways or not show up at all. More information can be found in the
+> [gdbundle README.md](https://github.com/memfault/gdbundle)
+
+## gdbundle Example Plugin
+
+No plugin manager introduction would be complete without an example plugin. I've
+created a "Hello World" plugin for gdbundle called
+[`gdbundle-example`](https://github.com/memfault/gdbundle-example). Let's dig in
+and see how it's built.
+
+### Structure
+
+Below is the directory structure of our example plugin that works with GDB and
+LLDB.
+
+```
+├── README.md
+├── gdbundle_example
+│   ├── __init__.py
+│   ├── gdb_loader.py
+│   ├── lldb_loader.py
+│   └── scripts
+│       ├── example_gdb.gdb
+│       ├── example_gdb.py
+│       └── example_lldb.py
+└── pyproject.toml
+```
+
+Simple, right? I've opted to use `pyproject.toml` instead of `setup.py` because
+I wanted to embrace the future of standardized packaging, but note that these
+packages will work with Python 2 and 3, so it shouldn't matter.
+
+### How It Works
+
+Let's look at the code in `gdbundle_example/gdb_loader.py`
+
+```python
+import gdb
+import os
+
+PACKAGE_DIR = os.path.dirname(__file__)
+
+SCRIPT_PATHS = [
+    [PACKAGE_DIR, 'scripts', 'example_gdb.gdb'],
+    [PACKAGE_DIR, 'scripts', 'example_gdb.py']
+]
+
+def _abs_path(path):
+    return os.path.abspath(os.path.join(*path))
+
+def gdbundle_load():
+    for script_path in SCRIPT_PATHS:
+        gdb.execute("source {}".format(_abs_path(script_path)))
+```
+
+Notice the `gdbundle_load` function which `gdbundle` will call directly. When
```

Comment:
I guess gdbundle could also call `get_scripts()` and do this itself? Did you opt not to do that so the script writer has more flexibility w.r.t what happens during load?

Reply:
I changed the way plugins loaded the code when I realized 2 things: 

- With PyCortexMDebug , it was easier for the package to be loaded by Python `import ...` instead of just sourcing files. 
- With the distinction between LLDB and GDB, it made it easier to delegate it to the plugins. 

I do eventually want to come up with both options so that a hollistic configuration story is better, but lets first see if things take off.